controllers/update_config_map.go

```diff
@@ -81,20 +84,39 @@ func (u UpdateConfigMap) Reconcile(r *FoundationDBClusterReconciler, context ctx
 		return false, err
 	}
 
+	var errs []error
+	// We try to update all instances and if we observe an error we add it to the error list.
+	// TODO: should we try to update the instances concurrently?
 	for index := range instances {
 		instance := instances[index]
-		if instance.Metadata.Annotations[LastConfigMapKey] != configMapHash {
-			synced, err := r.updatePodDynamicConf(cluster, instance)
-			if !synced {
-				return synced, err
-			}
+		if instance.Metadata.Annotations[LastConfigMapKey] == configMapHash {
+			continue
+		}
 
-			instance.Metadata.Annotations[LastConfigMapKey] = configMapHash
-			err = r.PodLifecycleManager.UpdateMetadata(r, context, cluster, instance)
+		synced, err := r.updatePodDynamicConf(cluster, instance)
+		if !synced {
+			log.Info("Update dynamic Pod config", "namespace", configMap.Namespace, "cluster", cluster.Name, "processGroupID", instance.GetInstanceID(), "synced", synced)
 			if err != nil {
-				return false, err
+				errs = append(errs, err)
+			} else {
+				errs = append(errs, fmt.Errorf("processGroupID %s not synced", instance.GetInstanceID()))
 			}
+
+			continue
 		}
+
+		instance.Metadata.Annotations[LastConfigMapKey] = configMapHash
+		err = r.PodLifecycleManager.UpdateMetadata(r, context, cluster, instance)
+		if err != nil {
+			log.Info("Update Pod metadata", "namespace", configMap.Namespace, "cluster", cluster.Name, "processGroupID", instance.GetInstanceID(), "error", err)
+			errs = append(errs, err)
+		}
+	}
+
+	if len(errs) > 0 {
+		// If we return an error we don't requeue
+		// So we just return that we can't continue but don't have an error
+		return false, nil
```

Comment:
The way how our reconcile loop work is pretty weird (and we might want to change this). If we return an error we just ignore the requeue time 🤔

Reply:
Yeah, I think this is an oversight and we should add the requeueAfter time to the error handling path as well.

Reply:
I'll open an issue for this one.